src/board/novacustom/v540tnx/board.mk

```diff
@@ -0,0 +1,58 @@
+# SPDX-License-Identifier: GPL-3.0-only
+
+board-y += board.c
+board-y += gpio.c
+
+EC = ite
+CONFIG_EC_ITE_IT5570E = y
+
+# Enable eSPI
+CONFIG_BUS_ESPI = y
+
+# Use S0ix
+CFLAGS+=-DUSE_S0IX=1
+
+# Include keyboard
+KEYBOARD = 14in_83
+KEYMAP=combo_iso_ansi_micmute
+
+# Set keyboard LED mechanism
+KBLED=white_dac
+CFLAGS+=-DKBLED_DAC=2
+
+# Set discrete GPU I2C bus
+CFLAGS+=-DI2C_DGPU=I2C_1
+
+# Set battery I2C bus
+CFLAGS += -DI2C_SMBUS=I2C_4
+
+# Set touchpad PS2 bus
+CFLAGS += -DPS2_TOUCHPAD=PS2_3
+
+# Set USB-PD I2C bus
+CFLAGS+=-DI2C_USBPD=I2C_1
+
+# Set smart charger parameters
+CHARGER = oz26786
+CFLAGS += \
+	-DCHARGER_ADAPTER_RSENSE=5 \
+	-DCHARGER_BATTERY_RSENSE=10 \
+	-DCHARGER_CHARGE_CURRENT=1536 \
+	-DCHARGER_CHARGE_VOLTAGE=17600 \
+	-DCHARGER_INPUT_CURRENT=6500
+
+# Set CPU power limits in watts
+# AC: power rating of the included AC adapter
+# DC: battery discharge rate (assume 1C rating??)
+CFLAGS += \
+	-DPOWER_LIMIT_AC=180 \
+	-DPOWER_LIMIT_DC=80
```

Comment:
These values are used to set PL3, PL4 and PsysPL2 in power_peci_limit. While PsysL2 and PL3 are fine (because PL3 becomes PsysPL3 if Psys is enabled), PL4 is wrong. PL4 is the maximum instantaneous electrical power drawn by the SoC, not by the whole system. So If compared to the iGPU variant with power limit of 90, it means that the same SoC can have double the max instantaneous power...  It may be one of the reasons of overheating.

Now, the comment says `// Cap PL4 to supply wattage if needed`, but actually, what happens is that PL4 is updated unconditionally based on AC brick capabilities. It does not read the current PL4 value and check if it needs to be capped. Also the MTL SOC PL4 is supposed to be 120W per coreboot code. So what will happen here:

1. 180W AC - 15% safety margin (27W): 153W
2. Bat boost ? 153W + 80W (233W)  : 153W
3. PL4 = 
   a) Bat boost: ( 233W > 180W ) ? 180W : 233W. So the value is 180W
   b) No bat boost: (153W > 180W) ? 180W : 153W. SO the value is 153W
   
In both cases it is above what Intel "allows" by default (120W).

So what we should or want to do here is to leverage PsysPmax or PMAX (platform scoped, not SOC/package scoped). PsysPmax = PL4 + ROP (Rest of Platform). We would like to cap the PsysPmax at the charger capability of 180W/153W (without bat boost).

The ROP power consumption can be dynamically adjusted by using the PL4 offset mechanism (settable via PECI as well). So we could leverage it to allocate power for dGPU. PL4 offset causes the SOC PL4 to be reduced by the amount specified by the PL4 offset. E.g. If we set the PL4 for SOC to be 120W as Intel recommends, it leaves 60W (with bat boost) for dGPU and ROP. By increasing PL4 offset we can decrease the SOC power to let dGPU spread the wings, e.g. PL4 offset = 40W it means, PL4 will be 80W and ROP + dGPU will have 100W.

Reply:
But you actually don't have to use DTT at all here. Also do we have proofs that this PL4 is not the cause?

Reply:
> Secondly, I don't see any policies for DTT in dGPU devicetree in coreboot. So how DTT knows what to do @mkopec ?

This is handled automatically without any firmware configuration. See doc #729152 @miczyg1 

I saw that with the default policy the dGPU would be auto-enumerated and power share would be applied automatically.

Reply:
I understand the doc a bit differently. Slide 12 already says that you have to configure some policy. Also I never expect "automatic and "default" to be always correct in firmware world.

Reply:
Okay, so the power limits work in a manner "lowest wins". Basically ,the PL4 will always be correct